test/role.test.js

```diff
@@ -511,6 +554,24 @@ describe('role model', function() {
       });
     });
 
+    it('should resolve user by id using custom user principalType', function(done) {
+      ACL.resolvePrincipal('User', user.id, function(err, u) {
+        if (err) return done(err);
+
+        expect(u.id).to.eql(user.id);
+
+        done();
+      });
+    });
+
+    it('should throw error when principalType is incorrect', function(done) {
+      ACL.resolvePrincipal('incorrectPrincipalType', user.id, function(err, u) {
+        assert(err);
```

Comment:
Please make the assertion more specific, verify that you got the expected error. The difficult part is how to write the check in such way, that it's not overly specific and won't break whenever the error message changes, but at the same time fails if a different error is reported.

In some other places of LoopBack, we started to use `err.code` with a machine-readable string that can be processed by clients, maybe we can do the same here?

```js
expect(err).to.have.property('code', 'INVALID_PRINCIPAL_TYPE');
```

Thoughts?